loki/transformations/dependency.py

```diff
@@ -16,96 +16,107 @@ class DuplicateKernel(Transformation):
 
     creates_items = True
 
-    def __init__(self, kernels=None, duplicate_suffix='duplicated',
+    reverse_traversal = True
+
+    def __init__(self, duplicate_kernels=None, duplicate_suffix='duplicated',
                  duplicate_module_suffix=None):
         self.suffix = duplicate_suffix
         self.module_suffix = duplicate_module_suffix or duplicate_suffix
-        print(f"suffix: {self.suffix}")
-        print(f"module_suffix: {self.module_suffix}")
-        self.kernels = tuple(kernel.lower() for kernel in as_tuple(kernels))
+        self.duplicate_kernels = tuple(kernel.lower() for kernel in as_tuple(duplicate_kernels))
+
+    def _create_duplicate_items(self, successors, item_factory, config):
```

Comment:
I feel like this should be a (more generic) utility that lives outside of this transformation (at least in the future).